master/internal/command/events.go

```diff
@@ -88,6 +135,43 @@ func (e *eventManager) Receive(ctx *actor.Context) error {
 				child.Stop()
 			}
 		}
+		e.processNewLogEvent(ctx, msg)
+
+	case webAPI.LogsRequest:
+		if ctx.Sender() == nil {
+			return errors.New(ctxMissingSender)
+		}
+		ctx.Respond(true)
+
+		total := countNonNullRingValues(e.buffer)
+		offset := webAPI.EffectiveOffset(msg.Offset, total)
+		msg.Offset = offset
+
+		// Stream existing matching entries.
+		matchingEvents := e.getMatchingEvents(msg)
+		for _, ev := range matchingEvents {
+			logEntry := eventToLogEntry(ev)
+			if logEntry != nil {
+				ctx.Tell(ctx.Sender(), *logEntry)
+			}
+		}
+
+		limitMet := msg.Limit > 0 && len(matchingEvents) >= msg.Limit
+
+		if msg.Follow && !e.isTerminated && !limitMet {
+			e.logStreams[ctx.Sender()] = msg
+		} else {
+			ctx.Tell(ctx.Sender(), webAPI.CloseStream{})
+		}
+
+	case webAPI.CloseStream:
+		if ctx.Sender() == nil {
+			return errors.New(ctxMissingSender)
```

Comment:
make this a panic, after the infra ag meeting where we decided to use panics on internal logic errors.

Reply:
what type of errors should we panic on ? maybe also places you'd usually do sanity checks?

Reply:
We should panic on errors which are 100% internal logic failures.  Like the kind of errors where "this should literally never occur unless there's a bug in the system".